cups/dest.c

```diff
@@ -552,7 +552,7 @@ cupsConnectDest(
         uri = cups_dest_resolve(dest, uri, msec, cancel, cb, user_data);
     }
   }
-  else if ((uri = cupsGetOption("printer-uri-supported", dest->num_options, dest->options)) == NULL)
+  else
```

Comment:
I'm not sure this is correct since the "printer-uri-supported" value will be present if a queue is setup...

Reply:
Unfortunately we can't be sure if the queue will stay present until the next function is called - this way we call Create-Local-Printer even if the queue exists, to prolong its existence (since someone want to connect to it).

Tbh it is only prolonging the interval where user can successfully ask the queue - now I think more stable solution and future one would be to incorporate connection+local printer creation into something like ```cupsDoRequest()```, but instead of http struct it would accept destination - the connection and printer would be created in the function, so it would prevent temp queue removal in the meantime...

Reply:
But permanent queues also have printer-uri-supported. We need to only try creating the local (temporary) queue if it is a temporary printer or one without a printer-uri. And ideally cupsd should let the client know when the temporary queue is scheduled to go away...

Reply:
An in ideal case - what about clients can use printer-state-change-date-time for checking when the queue is going away? Or we could write a public function for this, where the option value + 60s will be compared with current time, and return false if the current time is higher. Wdyt?

At the moment I implemented the former idea.